python_modules/libraries/dagster-shared/dagster_shared/libraries/__init__.py

```diff
@@ -32,17 +32,21 @@ def check_dagster_package_version(library_name: str, library_version: str) -> No
     # This import must be internal in order for this function to be testable
     from dagster_shared.version import __version__
 
+    parsed_current = parse_package_version(__version__)
     parsed_lib_version = parse_package_version(library_version)
     if parsed_lib_version.release[0] < 1:
-        if library_version != __version__:
+        if parsed_lib_version.release != parsed_current.release:
```

Comment:
https://packaging.pypa.io/en/latest/version.html

I think we may want to only allow skew in `post` specifically but wanted to throw up something quick for discussion to get back to what i was doing